### PR TITLE
feat: [CCM-8509]: Misc fixes in dashboard screen

### PR DESCRIPTION
**Pull request by jyotiarora-harness**
### Summary

Fixes:
1. Overflow layout 
2. Error handling
3. Empty state handling
4. Free edition banner handling

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 9 Issues 
* Code issue 9 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10403)

File: src/modules/75-ce/pages/bi-dashboards/BIDashboard.module.scss [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
 .header {
   padding: var(--spacing-medium) var(--spacing-large) !important;
   border-bottom: 1px solid var(--grey-200);
   background: var(--white) !important;
 }
 
 .pageBodyWrapper {
-  padding: 0 150px !important;
+  flex-grow: 1;
+  width: min(100%, 1155px);
```
Comment (jenil-harness) [inline src/modules/75-ce/pages/bi-dashboards/BIDashboard.module.scss:16]: Why this?

Comment (jyotiarora-harness) [reply src/modules/75-ce/pages/bi-dashboards/BIDashboard.module.scss:16]: Due to the min-width changes in core layout overflow was happening in the screen, did the handling for the same

File: src/modules/75-ce/pages/bi-dashboards/BIDashboard.tsx [hunk 2]
```diff
@@ -172,4 +166,32 @@ const BIDashboard: React.FC = () => {
   )
 }
 
+const BIDashboard: React.FC = () => {
+  const { getString } = useStrings()
+  const { module } = useModuleInfo()
+  const { licenseInformation } = useLicenseStore()
+  const moduleName: ModuleName = module ? moduleToModuleNameMapping[module] : ModuleName.COMMON
+  const isEnterpriseEdition = isEnterprisePlan(licenseInformation, moduleName)
```
Comment (jenil-harness) [inline src/modules/75-ce/pages/bi-dashboards/BIDashboard.tsx:174]: No need to default to Common here.

File: src/modules/75-ce/pages/perspective-list/PerspectiveListPage.tsx [hunk 3]
```diff
@@ -226,7 +226,7 @@ interface NoDataPerspectivePageProps {
   showConnectorModal?: boolean
 }
 
-const NoDataPerspectivePage: (props: NoDataPerspectivePageProps) => JSX.Element = ({ showConnectorModal }) => {
+export const NoDataPerspectivePage: (props: NoDataPerspectivePageProps) => JSX.Element = ({ showConnectorModal }) => {
```
Comment (jenil-harness) [inline src/modules/75-ce/pages/perspective-list/PerspectiveListPage.tsx:229]: We should move it to a common place

File: src/modules/75-ce/pages/bi-dashboards/BIDashboard.tsx (hunk 2 above)
Comment (jyotiarora-harness) [reply src/modules/75-ce/pages/bi-dashboards/BIDashboard.tsx:174]: module can be undefined as well, so needs to do the default handling, should I use the default value as `ModuleName.CE` here?

File: src/modules/75-ce/pages/bi-dashboards/__tests__/BIDashboard.test.tsx [hunk 4]
```diff
@@ -23,14 +31,39 @@ jest.mock('services/ce', () => ({
...
+    const isEnterpriseEdition = true
+    const responseState = {
+      executeQuery: () => {
+        return fromValue({})
+      }
+    }
+    const { container } = render(
+      <TestWrapper pathParams={params}>
+        <Provider value={responseState as any}>
+          {isEnterpriseEdition ? <BIDashboardData /> : <LevelUpBanner message={'dashboards.upgrade'} />}
```
Comment (rajarshee-harness) [inline src/modules/75-ce/pages/bi-dashboards/__tests__/BIDashboard.test.tsx:43]: Why do we have this condition when `isEnterpriseEdition` is always true?

File: src/modules/75-ce/pages/bi-dashboards/BIDashboard.tsx [hunk 5]
```diff
@@ -172,4 +166,32 @@ const BIDashboard: React.FC = () => {
...
+
+  const banner = <LevelUpBanner message={getString('dashboards.upgrade')} />
```
Comment (rishik30) [inline src/modules/75-ce/pages/bi-dashboards/BIDashboard.tsx:176]: what's the need of storing it in a variable? Are we using it elsewhere too?

File: src/modules/75-ce/pages/perspective-list/PerspectiveListPage.tsx (hunk 3 above)
Comment (rishik30) [reply src/modules/75-ce/pages/perspective-list/PerspectiveListPage.tsx:229]: +1

**Review by rishik30 — CHANGES_REQUESTED** [review]

**Review by rajarshee-harness — DISMISSED** [review]

**Review by rishik30 — APPROVED** [review]

**Review by rishik30 — APPROVED** [review]

**Review by rajarshee-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**